lib/floki/selector.ex

```diff
@@ -103,15 +103,17 @@ defmodule Floki.Selector do
   defp id_match?(%HTMLNode{attributes: []}, _), do: false
   defp id_match?(%HTMLNode{type: :pi}, _), do: false
 
-  defp id_match?(%HTMLNode{attributes: attributes}, id) do
-    Enum.any?(attributes, fn attribute ->
-      case attribute do
-        {"id", ^id} -> true
-        _ -> false
-      end
-    end)
+  defp id_match?(%HTMLNode{attributes: attributes}, id) when is_list(attributes) do
+    id_attr_value = :proplists.get_value("id", attributes, nil)
+    id_attr_value == id
+  end
+
+  defp id_match?(%HTMLNode{attributes: %{"id" => id_attr_value}}, id) do
```

Comment:
Even better!
It's fine this way, but just as a complement: you could match in the function definition as well.

```suggestion
  defp id_match?(%HTMLNode{attributes: %{"id" => id}}, id), do: true
```
If you use the same binding name, it is going to evaluate only if they are the same value.

Reply:
I always forget about this pattern matching feature, a754b67fbc5b9a241efbda68704f94b0c67f98ff